waspc/src/Wasp/Generator/SdkGenerator/Server/JobGenerator.hs

```diff
@@ -42,29 +52,81 @@ genIndexTs jobs = return $ C.mkTmplFdWithData tmplFile tmplData
           "jobName" .= jobName
         ]
 
-genJobType :: (String, Job) -> Generator FileDraft
-genJobType (jobName, job) =
+genJob :: (String, Job) -> Generator FileDraft
+genJob (jobName, job) =
   return $
     C.mkTmplFdWithDstAndData
       tmplFile
       dstFile
       $ Just tmplData
   where
-    tmplFile = [relfile|server/jobs/_jobTypes.ts|]
+    tmplFile = [relfile|server/jobs/_job.ts|]
     dstFile = [reldir|server/jobs|] </> fromJust (SP.parseRelFile $ jobName ++ ".ts")
     tmplData =
       object
-        [ "typeName" .= toUpperFirst jobName,
-          "jobExecutorTypesImportPath" .= SP.fromRelFileP jobExecutorTypesImportPath,
-          "entities" .= maybe [] (map (makeJsonWithEntityData . AS.refName)) (J.entities job)
+        [ "jobName" .= jobName,
+          "typeName" .= toUpperFirst jobName,
+          "jobExecutorImportPath" .= SP.fromRelFileP jobExecutorImportPath,
+          "entities" .= maybe [] (map (makeJsonWithEntityData . AS.refName)) (J.entities job),
+          -- NOTE: You cannot directly input an Aeson.object for Mustache to substitute.
+          -- This is why we must get a text representation of the object, either by
+          -- `Aeson.Text.encodeToLazyText` on an Aeson.Object, or `show` on an AS.JSON.
+          "jobSchedule" .= Aeson.Text.encodeToLazyText (fromMaybe Aeson.Null maybeJobSchedule),
```

Comment:
Hm, I'm not sure about this. I remember doing it for generating query and action files. We usually send `Aeson.Value`, but those can be objects too, right? I'm not sure what the difference is.

Reply:
> Hm, I'm not sure about this. I remember doing it for generating query and action files. We usually send Aeson.Value, but those can be objects too, right? I'm not sure what the difference is.

This is old code and I didn't want to change it alongside all the other changes in this PR. We can rewrite this later, but it also works okay now 🤷